bin/burden

```diff
@@ -231,24 +258,71 @@ else
 	UTILS_DIR=$UTILS_DIR/utils
 fi
 
+#
+# Exclusive locking routines
+#
+# Given bash does not have a locking mechism built in, for code blocks
+# we need to have our own.
+#
+# To do this:
+#  Locking exclusive: Attempt to make the directory $gl_lock.  If the mkdir
+#     fails, then sleep for 4 seconds, and repeat.  If mkdir passes, then set 
+#     $gl_have_locked to be 0, and break from the loop.
+# Unlocking exclusive: Simply remove the dir $gl_lock and set $gl_have_locked
+#     to be 0.
+#
+# $gl_have_locked is to make sure we do not attempt to remove the lock if we do not
+#     own it.  The cleanup_and_exit routine simply attempts to unlock it everytime,
+#     this prevents that from happening if we do not own the lock.
+#
+gl_lock=${gl_top_dir}/gl_lock
+gl_have_locked=0
+gl_debug_lock=0
+
+gl_lock_exclusive()
+{
+	if [[ $gl_debug_lock -eq 1 ]]; then
+		echo lock $1
```

Comment:
So the arg passed is purely for debug?  It'd be good to note that, though I'm still wondering if this is going to cause an issue at some point down the road but maybe it's good enough for now?

Reply:
Have changed it so we report the calling function and line number via
echo ${FUNCNAME[1]} ${BASH_LINENO[0]}
nothing needs to be passed into the lock functions.